src/sqlparser/src/parser.rs

```diff
@@ -2360,17 +2360,32 @@ impl Parser {
     }
 
     pub fn parse_explain(&mut self, describe_alias: bool) -> Result<Statement, ParserError> {
-        let analyze = self.parse_keyword(Keyword::ANALYZE);
-        let verbose = self.parse_keyword(Keyword::VERBOSE);
-        let trace = self.parse_keyword(Keyword::TRACE);
-
+        let mut analyze = false;
+        let mut distsql = false;
+        let mut verbose = false;
+        let mut trace = false;
+        while let Some(keyword) = self.parse_one_of_keywords(&[
+            Keyword::ANALYZE,
+            Keyword::DISTSQL,
+            Keyword::VERBOSE,
+            Keyword::TRACE,
+        ]) {
+            match keyword {
+                Keyword::ANALYZE => analyze = true,
+                Keyword::DISTSQL => distsql = true,
+                Keyword::VERBOSE => verbose = true,
+                Keyword::TRACE => trace = true,
```

Comment:
```rs
enum ExplainMode {
  Analyze,
  DistSql,
  Verbose,
  Trace
}
```

Reply:
the explain mode is not a “one of them” they are independent flag

Reply:
Does that mean we can write a EXPLAIN like:

```sql
EXPLAIN VERBOSE ANALYZE TRACE DISTSQL SELECT * FROM t
```

Reply:
- ANALYZE: the SQL will really executed, but in our system, we have not implemented it now
- VERBOSE: output more information about the final plan
- TRACE: not only output the final plan, will output the optimization phases
- DISTSQL: the final plan is the fregments.

Reply:
According to https://www.postgresql.org/docs/current/sql-explain.html, we can add serveral options together.

Reply:
I think the standard pg (cockroach as well) syntax is like: 

```sql
explain (verbose, analyze) select 1;
```

Postgres additionally supports `explain (verbose true, analyze true) select 1`, though I think the boolean parameter of each option is redundant. 

The implementation in this pr doesn't align with both. Furthermore, according to our current code, the option `analyze` is unused. We need to refine the command definition and make the syntax clear.

Reply:
Agree that we should clear out unsupported parameters.

Reply:
To be more clear, PG syntax is: (Note the second line)
```
EXPLAIN [ ( option [, ...] ) ] statement
EXPLAIN [ ANALYZE ] [ VERBOSE ] statement
```
i.e.,
```sql
# valid
explain analyze verbose select 1;
explain (summary) select 1;

# invalid
explain summary select 1;
```
While crdb doesn't have the second syntax, i.e., parentheses are always required.

Reply:
Maybe PG keeps the second for compatibility:

> Only the ANALYZE and VERBOSE options can be specified, and only in that order, without surrounding the option list in parentheses. Prior to PostgreSQL 9.0, the unparenthesized syntax was the only one supported. It is expected that all new options will be supported only in the parenthesized syntax.

I would +1 for always requiring parentheses.

Reply:
BTW, fun fact in PG 😄:
```sql
explain (summary, summary) select 1;
explain (summary true, summary false) select 1;
```

Reply:
https://www.postgresql.org/docs/current/sql-explain.html

> boolean
> Specifies whether the selected option should be turned on or off. You can write TRUE, ON, or 1 to enable the option, and FALSE, OFF, or 0 to disable it. The boolean value can also be omitted, in which case TRUE is assumed.
>